### PR TITLE
Search: Add search plan to my-plans page

### PR DESCRIPTION
**Pull request by gibrown**
Changes:
- Adds the search plan to the /my-plans page.
- Removes the feature card for search for pro plans (at some point we will probably remove it)
- Don't show the upgrade nudge if they already have a plan

Screen Shot 2020-03-24 at 12 22 30 PM

To test:
- Go to wp-admin/admin.php?page=jetpack#/my-plan
- Use "Dev tools" at the bottom of the page to try different plans out (unfortunately you can't do combos: https://github.com/Automattic/jetpack/issues/15071)

Fixes https://github.com/Automattic/jetpack/issues/15070

### CODE REVIEW

**Comment by keoshi** [comment]
I believe the existing pattern (for JP Backup) is to reuse the plan icon, which I've added on https://github.com/Automattic/wp-calypso/pull/40382

**Review by keoshi — APPROVED** [review]
LGTM!

**Review by jsnmoon — CHANGES_REQUESTED** [review]
I don't think this is working as intended. My site has a Jetpack Free plan and a Jetpack Search product purchased via the Checkout flow via Calypso.

With this change, the Instant Search card is missing:

Screen Shot 2020-03-25 at 4 49 27 PM

When I manually set my plan to Search, I get the following:

Screen Shot 2020-03-25 at 4 50 57 PM

**Review by jsnmoon — APPROVED** [review]
I just pushed a commit that fixes the bugs mentioned in my previous review. Please double-check my changes before merging!

File: _inc/client/my-plan/index.jsx [hunk 1]
```diff
@@ -7,52 +7,52 @@ import { connect } from 'react-redux';
...
-
-	render() {
-		return (
-			<div>
-				<QuerySite />
-				{ this.renderContent() }
-			</div>
-		);
+export function MyPlan( props ) {
+	const hasSearchProduct = !! find( props.purchases, purchase =>
```
Comment (jsnmoon) [inline _inc/client/my-plan/index.jsx]: I knew I forgot something; this needs an `import { find } from 'lodash';` for it to work properly.

**Pull request merged**